### PR TITLE
Add tutorial to lesson migration command

### LINKED ISSUE #3043: Convert Tutorials to Lessons
Create a CLI command that takes a list of tutorial URLs and converts them to Lessons, by changing the post type.

Run it in prod to convert the Tutorials in this [sheet](https://docs.google.com/spreadsheets/d/1D_caQxJs55eIinGTlIb7du_vE5FBxn4ri7pAWI5K2RE/edit?gid=0#gid=0).

### PR DESCRIPTION
**Pull request by adamwoodnz**
See #2393 
Closes https://github.com/WordPress/Learn/issues/3043

Adds a CLI command to mu-plugins for converting tutorials to lessons, by changing the post type. Takes a single post URL, or file with a list of URLs, which will be created from this [sheet](https://docs.google.com/spreadsheets/d/1D_caQxJs55eIinGTlIb7du_vE5FBxn4ri7pAWI5K2RE/edit?gid=0#gid=0).

By default runs in dry mode, which prints the details of the post(s).

I plan to update the sync script to copy this into `wporg/public_html/wp-content/mu-plugins/pub`

### CODE REVIEW

File: wp-content/mu-plugins/pub/wporg-learn-cli.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	/**
+	 * Converts tutorials to lessons based on the provided URL or file.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <source>
+	 * : The URL of the tutorial post to convert, or path to a file containing URLs (one per line)
+	 *
+	 * [--live]
+	 * : Actually perform the conversion (default is dry-run)
```
Comment (pkevan) [inline wp-content/mu-plugins/pub/wporg-learn-cli.php:20]: in most cases having a `dry-run` variable is how it's normally done, but this is a minor nitpick and not a blocker.

File: wp-content/mu-plugins/pub/wporg-learn-cli.php [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+			$this->process_url( $url, $is_dry_run );
+		}
+	}
+
+	/**
+	 * Process a single URL.
+	 */
+	private function process_url( $url, $is_dry_run ) {
+		// Get post ID from URL
+		$post_id = url_to_postid( $url );
```
Comment (pkevan) [inline wp-content/mu-plugins/pub/wporg-learn-cli.php:63]: `url_to_postid` is an expensive operation, so it should be considered if running through a large amount of items (which I think this instance isn't).

File: wp-content/mu-plugins/pub/wporg-learn-cli.php [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+			if ( ! file_exists( $source ) ) {
+				WP_CLI::error( "File not found: $source" );
+				return;
+			}
+			$urls = array_filter( explode( "\n", file_get_contents( $source ) ) );
+		} else {
+			$urls = array( $source );
+		}
+
+		foreach ( $urls as $url ) {
```
Comment (pkevan) [inline wp-content/mu-plugins/pub/wporg-learn-cli.php:47]: some consideration should be made for memory expensive operations when looping through an indeterminate number of items

**Review by pkevan — COMMENTED** [review]
most of my comments are minor given the extend of the command being used (less than 100 items)

File: wp-content/mu-plugins/pub/wporg-learn-cli.php (hunk 2 above)
Comment (adamwoodnz) [reply wp-content/mu-plugins/pub/wporg-learn-cli.php:63]: TIL

**Comment by adamwoodnz** [comment]
> most of my comments are minor given the extend of the command being used (less than 100 items)

All fair, thanks!

**Pull request merged**